test/specs/events.spec.js

```diff
@@ -1,56 +1,224 @@
+        animation: false,
+        plugins: {
+          legend: false,
+          annotation: {
+            annotations: [
+              {
+                type: 'line',
+                scaleID: 'y',
+                value: 3
+              },
+              {
+                type: 'line',
+                scaleID: 'y',
+                value: 3,
+                enter: () => true
+              },
+              {
+                type: 'line',
+                scaleID: 'y',
+                value: 3
+              }
+            ]
+          }
         }
-      },
-      plugins: {
-        legend: false,
-        annotation: {
-          annotations: [
-            {
-              type: 'line',
-              scaleID: 'y',
-              value: 3
-            },
-            {
-              type: 'line',
-              scaleID: 'y',
-              value: 3,
-              enter: () => true
-            },
-            {
-              type: 'line',
-              scaleID: 'y',
-              value: 3
-            }
-          ]
+      }
+    };
+    it('should detect listened and moveListened state set to true', function() {
+      const chart = window.acquireChart(chartConfig);
+      const Annotation = window['chartjs-plugin-annotation'];
+      const state = Annotation._getState(chart);
+      expect(state.listened).toBe(true);
+      expect(state.moveListened).toBe(true);
+    });
+  });
+
+  describe('check', function() {
+    const chartConfig = {
+      type: 'scatter',
+      options: {
+        animation: false,
+        scales: {
+          x: {
+            display: false,
+            min: 0,
+            max: 10
+          },
+          y: {
+            display: false,
+            min: 0,
+            max: 10
+          }
+        },
+        plugins: {
+          legend: false,
+          annotation: {
+          }
         }
       }
-    },
-  };
+    };
+    const options = {
+      type: 'box',
+      id: 'test',
+      xScaleID: 'x',
+      yScaleID: 'y',
+      xMin: 2,
+      yMin: 2,
+      xMax: 4,
+      yMax: 4
+    };
+    const x0y0 = {x: 0, y: 0};
+    const center = function(chart) {
+      const xCenter = options.xMax - options.xMin;
+      const yCenter = options.yMax - options.yMin;
+      const xScale = chart.scales.x;
+      const yScale = chart.scales.y;
+      return {x: xScale.getPixelForValue(xCenter), y: yScale.getPixelForValue(yCenter)};
+    };
 
-  it('should not throw any exception', function() {
+    const pluginOpts = chartConfig.options.plugins.annotation;
 
-    const chart = window.acquireChart(chartConfig);
-    const Annotation = window['chartjs-plugin-annotation'];
-    const state = Annotation._getState(chart);
+    [pluginOpts, options].forEach(function(targetOptions) {
+      it('should not detect any move events (because no callback is set)', function(done) {
+        const enterSpy = jasmine.createSpy('enter');
+        const leaveSpy = jasmine.createSpy('leave');
```

Comment:
These spies are not attached to anything, and thus will never be called?
You can not test if a callback is called when it does not exist :)

I think there is no good way of testing this, and thus this test can be removed.

Reply:
This test case is testing that only click are set and test what is happening when a move event is coming (all `else` cases). 
I can remove it but I'm going to check which part of the code will remain uncovered by tests